### PR TITLE
doc/cephadm: add a terminology section

### PR DESCRIPTION
**Pull request by jschmid1**
Signed-off-by: Joshua Schmid <jschmid@suse.de>

Opening the PR to gather feedback. This is not ready yet and is missing proper integration in the document as well as proper cross section linking.

### CODE REVIEW

**Comment by sebastian-philipp** [comment]
can you unify this with https://docs.ceph.com/docs/master/mgr/orchestrator/#orchestrator-cli ?

File: doc/cephadm/terminology.rst [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+============
+Termninology
+============
+
+
+`Cephadm` introduces quite a few new terms that need clear demarcations. This article is supposed to go over them and
+explain what they are all about.
+
+
+cephadm (The manager module)
```
Comment (sebastian-philipp) [inline doc/cephadm/terminology.rst:12]: `mgr/cephadm`?

File: doc/cephadm/terminology.rst [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+============================
+
+This is the thing we're talking about when someone is referring to `cephadm`. It is implemented
+as a `mgr module` and is currently the most elaborate orchestrator for ceph.
+Cephadm contains most of the orchestration logic including the `Scheduling` algorithms.
+It's often referred to `mgr/cephadm` to distinguish it from `cephadm` (The tool).
+
+
+cephadm (The daemon tool)
```
Comment (sebastian-philipp) [inline doc/cephadm/terminology.rst:21]: can we link to https://docs.ceph.com/docs/master/man/8/cephadm/ here?

File: doc/cephadm/terminology.rst [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+The term `Services` refers to a group of `Daemons` (ref) that make up some sort of complete entitiy(rewrite).
+On the example of a Rados Gateway that is configured to have three instances running at all times, we'd speak
+of the the `Rados Gateway Service`
+
+Service Specifications
+======================
+
+A Service Specification declaratively describes `Services` (ref). These mostly contain meta information about
+a Service, like name, service_type, placement and more. The only way to alter a `Service` is via it's specification.
```
Comment (sebastian-philipp) [inline doc/cephadm/terminology.rst:39]:
```suggestion
a Service, like ``service_id``, ``service_type``, ``placement`` and more. The only way to alter a `Service` is via it's specification.
```
should link to https://docs.ceph.com/docs/master/mgr/orchestrator/#service-specification

File: doc/cephadm/terminology.rst [hunk 4]
```diff
@@ -0,0 +1,49 @@
+.. _cephadm-termnilogy:
+
+============
+Termninology
+============
+
+
+`Cephadm` introduces quite a few new terms that need clear demarcations. This article is supposed to go over them and
+explain what they are all about.
```
Comment (asettle) [inline doc/cephadm/terminology.rst:9]: Demarcations makes sense, but it's not great for such a global audience. Let's go for something like....

File: doc/cephadm/terminology.rst [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+Cephadm contains most of the orchestration logic including the `Scheduling` algorithms.
+It's often referred to `mgr/cephadm` to distinguish it from `cephadm` (The tool).
+
+
+cephadm (The daemon tool)
+=========================
+
+This is the tool that operates on a host level and contains logic on how to deploy individual daemons like monitors, managers
+etc. This tool is usually not being used by the end-user but is rather used by the orchestrator module `cephadm`.
+To separate it from it's counterpart, the manager module `cephadm`, it's often referred to `bin/cephadm`.
```
Comment (asettle) [inline doc/cephadm/terminology.rst:26]: This spacing is all a bit odd. I'd recommend wrapping at 80 chars, just as a general rule for reviewing and reading :)

File: doc/cephadm/terminology.rst [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+This is the thing we're talking about when someone is referring to `cephadm`. It is implemented
+as a `mgr module` and is currently the most elaborate orchestrator for ceph.
+Cephadm contains most of the orchestration logic including the `Scheduling` algorithms.
+It's often referred to `mgr/cephadm` to distinguish it from `cephadm` (The tool).
+
+
+cephadm (The daemon tool)
+=========================
+
+This is the tool that operates on a host level and contains logic on how to deploy individual daemons like monitors, managers
```
Comment (asettle) [inline doc/cephadm/terminology.rst:24]:
```suggestion
This is the tool that operates on a host level and contains logic on how to deploy individual
daemons like monitors and managers.

File: doc/cephadm/terminology.rst [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+etc. This tool is usually not being used by the end-user but is rather used by the orchestrator module `cephadm`.
```
Comment (asettle) [inline doc/cephadm/terminology.rst:25]:
```suggestion
This tool is usually not used by the end-user but rather by the orchestrator module `cephadm`.

File: doc/cephadm/terminology.rst (hunk 5 above)
Comment (asettle) [inline doc/cephadm/terminology.rst:26]:
```suggestion
To separate it from its counterpart, `mgr/cephadm`, it's often referred to `bin/cephadm`.

File: doc/cephadm/terminology.rst [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+To separate it from it's counterpart, the manager module `cephadm`, it's often referred to `bin/cephadm`.
+
+Services
+========
+
+The term `Services` refers to a group of `Daemons` (ref) that make up some sort of complete entitiy(rewrite).
```
Comment (asettle) [inline doc/cephadm/terminology.rst:31]:
```suggestion
The term `Services` refers to a group of `Daemons` (ref) that make up some sort of complete entity (rewrite).

Comment (asettle) [reply doc/cephadm/terminology.rst:31]: What is `(ref)`?

File: doc/cephadm/terminology.rst [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+On the example of a Rados Gateway that is configured to have three instances running at all times, we'd speak
+of the the `Rados Gateway Service`
```
Comment (asettle) [inline doc/cephadm/terminology.rst:33]: I'm not sure what this means at all. What example of Rados Gateway? Could you link to it? Does this example have "three instances running at all times"? 
What do you mean by "we'd speak of the Rados Gateway Service"? Speak how? Refer to? Explain? This is relevant to?

Comment (asettle) [inline doc/cephadm/terminology.rst:33]:
```suggestion
of the `Rados Gateway Service`

File: doc/cephadm/terminology.rst [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+Service Specifications
```
Comment (asettle) [inline doc/cephadm/terminology.rst:35]: You've used sentence case in titles before, not capital case.

Comment (asettle) [inline doc/cephadm/terminology.rst:35]:
```suggestion
Service specifications

File: doc/cephadm/terminology.rst [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+======================
+
+A Service Specification declaratively describes `Services` (ref). These mostly contain meta information about
```
Comment (asettle) [inline doc/cephadm/terminology.rst:38]: Again, what's `(ref)`? Is it a missing reference point?

Comment (asettle) [inline doc/cephadm/terminology.rst:38]:
```suggestion
A Service Specification describes the `Services` (ref). These services contain meta information about

File: doc/cephadm/terminology.rst (hunk 3 above)
Comment (asettle) [reply doc/cephadm/terminology.rst:39]: Not `it's` - this means "it is". Please use `its` :)

File: doc/cephadm/terminology.rst [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+a Service, like name, service_type, placement and more. The only way to alter a `Service` is via it's specification.
+More information about the Specification can be found here. TODO: Link to Service Specification page.
+
+
+Daemons
+=======
+
+Daemons are the building blocks of Services. A Daemon is a single instance of a `Service` like a `ceph-mgr`, `ceph-mon`
```
Comment (asettle) [inline doc/cephadm/terminology.rst:46]: Daemons are the building blocks of Services. A daemon is a single instance of a `Service`. For example the `ceph-mgr`, `ceph-mon`

**Comment by jschmid1** [comment]
> can you unify this with https://docs.ceph.com/docs/master/mgr/orchestrator/#orchestrator-cli ?

Thanks for the reviews! I'll include my changes in this ^ document.

File: doc/cephadm/terminology.rst (hunk 3 above)
Comment (jschmid1) [reply doc/cephadm/terminology.rst:39]: > should link to https://docs.ceph.com/docs/master/mgr/orchestrator/#service-specification

As stated in the PR body, references are not included yet.

**Comment by jschmid1** [comment]
upon closer look we have the information already; just not in one place..
Closing this for now. Sorry for the noise

**Comment by sebastian-philipp** [comment]
> upon closer look we have the information already; just not in one place..
> Closing this for now. Sorry for the noise

nonono! this PR is really useful!

**Comment by sebastian-philipp** [comment]
> > can you unify this with https://docs.ceph.com/docs/master/mgr/orchestrator/#orchestrator-cli ?
> 
> Thanks for the reviews! I'll include my changes in this ^ document.

got it.

**Pull request closed without merging**